cdk/function.py

```diff
@@ -0,0 +1,187 @@
+from aws_cdk import (
+    core,
+    aws_lambda,
+    aws_ec2 as ec2,
+    aws_cloudwatch as cloudwatch,
+    aws_logs as logs
+)
+from . import names
+
+class ZipFunction(core.Construct):
+
+    def __init__(self, scope: core.Construct, id: str,
+                 name,
+                 lambda_code_bucket,
+                 environment,
+                 timeout=30,
+                 vpc_id=None,
+                 security_group_id=None):
+        super().__init__(scope, id)
+
+        self.stack_name = core.Stack.of(self).stack_name
+        environment = {key: str(val) for key,val in environment.items() if val}
+
+        function_props = {
+            "function_name": f"{self.stack_name}-{name}",
+            "runtime": aws_lambda.Runtime.PYTHON_3_8,
+            "code": aws_lambda.Code.from_bucket(
+                bucket=lambda_code_bucket,
+                key=f"{self.stack_name}/{name}.zip"
+            ),
+            "handler": f"{name}.handler",
+            "timeout": core.Duration.seconds(timeout),
+            "environment": environment,
+            "log_retention": logs.RetentionDays.SIX_MONTHS
+        }
+
+        if vpc_id is not None and security_group_id is not None:
+            opencast_vpc = ec2.Vpc.from_lookup(self, "OpencastVpc", vpc_id=vpc_id)
+            opencast_security_group = ec2.SecurityGroup.from_security_group_id(
+                self, "OpencastSecurityGroup", security_group_id=security_group_id
+            )
+            function_props.update({
+                "vpc": opencast_vpc,
+                "security_groups": [opencast_security_group]
+            })
```

Comment:
I tried to do this only in the uploader function construct but i couldn't get the constructor override, `super().__init__` part to work properly.